### PR TITLE
Implement Legends Awakened

### PR DESCRIPTION
**Pull request by luongthomas**
Gen 4 set

### CODE REVIEW

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 1]
```diff
@@ -2675,189 +2971,192 @@ public enum LegendsAwakened implements LogicCardInfo {
...
             }
           }
           move "Double Kick", {
-            text "20× damage. Flip 2 coins. This attack does 20 damage times the number of heads."
+            text "20x damage. Flip 2 coins. This attack does 20 damage times the number of heads."
             energyCost L, C
             attackRequirement {}
             onAttack {
-              damage 0
+              flip {
```
Comment (ufodynasty) [inline src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:3014]: You need to flip 2 coins.

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 2]
```diff
@@ -2894,276 +3192,300 @@ public enum LegendsAwakened implements LogicCardInfo {
...
           }
         };
       case MAGNEZONE_LV_X_142:
-        return levelUp (this, from:"Magnezone", hp:HP140, type:METAL, retreatCost:4) {
+        return levelUp (this, from:"Magnezone", hp:HP140, type:M, retreatCost:4) {
           weakness R
           resistance P, MINUS20
           pokePower "Electric Trans", {
-            text "As often as you like during your turn , you may move a or Energy attached to 1 of your Pokémon to another of your Pokémon. This power can’t be used if Magnezone is affected by a Special Condition."
+            text "As often as you like during your turn (before your attack), you may move a Lightning or [M] Energy attached to 1 of your Pokémon to another of your Pokémon. This power can't be used if Magnezone is affected by a Special Condition."
```
Comment (ufodynasty) [inline src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:3396]: Lightning should be [L] here?

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 3]
```diff
@@ -2894,276 +3192,300 @@ public enum LegendsAwakened implements LogicCardInfo {
...
-        return levelUp (this, from:"Mesprit", hp:HP090, type:PSYCHIC, retreatCost:1) {
+        return levelUp (this, from:"Mesprit", hp:HP090, type:P, retreatCost:1) {
           weakness P
           move "Healing Look", {
             text "Remove 3 damage counters from each of your Benched Pokémon."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
+            attackRequirement {
               assert my.bench : "Your bench is empty"
```
Comment (ufodynasty) [inline src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:3417]: I think assert my.bench.findAll{it.numberOfDamageCounters} would be better here, as it prevents accidentally using this move to no effect.

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 4]
```diff
@@ -2894,276 +3192,300 @@ public enum LegendsAwakened implements LogicCardInfo {
...
                 heal 30, it
               }
             }
           }
           move "Supreme Blast", {
             text "200 damage. If you don't have Uxie LV.X and Azelf LV.X in play, this attack does nothing. Discard all Energy attached to Mesprit."
             energyCost P, P
-            attackRequirement {my.all.findAll{it.name == ("Uxie Lv.X")} && my.all.findAll{it.name == ("Uxie Lv.X")}}
+            attackRequirement {
+              my.all.findAll { it.name == ("Uxie Lv.X") } && my.all.findAll{ it.name == ("Uxie Lv.X") }
```
Comment (ufodynasty) [inline src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:3429]: Typo here. You checked for Uxie twice.

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 5]
```diff
@@ -2894,276 +3192,300 @@ public enum LegendsAwakened implements LogicCardInfo {
...
             }
           }
           move "Giga Burn", {
             text "120 damage. Discard all Energy attached to Mewtwo."
             energyCost P, P, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 120
+              discardAllSelfEnergy()
```
Comment (ufodynasty) [inline src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:3452]: this needs to be after damage

**Review by ufodynasty — COMMENTED** [review]
Thank you so much!

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 6]
```diff
@@ -304,7 +308,19 @@ public enum LegendsAwakened implements LogicCardInfo {
             energyCost P
             attackRequirement {}
             onAttack {
-
+              discardSelfEnergyInOrderTo(P)
```
[the remark this answers is not in the document]
Comment (luongthomas) [reply src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:311]: :(

File: src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy [hunk 7]
```diff
@@ -1115,6 +1290,11 @@ public enum LegendsAwakened implements LogicCardInfo {
             attackRequirement {}
             onAttack {
               damage 10
+              flip 1, {
+                damage 10
+              }, {
+                directDamage 10, self
```
[the remark this answers is not in the document]
Comment (luongthomas) [reply src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy:1296]: What about the [P] weakness when damaging itself?

**Comment by luongthomas** [comment]
Thank you for the thorough review @ufodynasty !!!

**Pull request merged**